packages/components/src/internal/components/AutoForm.tsx

```diff
@@ -0,0 +1,284 @@
+    const className = inputClasses.number ?? '';
+    const _value = value === null || value === undefined ? '' : value;
+    return (
+        <input
+            className={className}
+            id={id}
+            inputMode="numeric"
+            name={name}
+            pattern="[0-9]*"
+            placeholder={placeholder}
+            type="text"
+            value={_value}
+            onChange={_onChange}
+        />
+    );
+};
+
+const TextareaInput: FC<AutoFormFieldProps> = ({ field, id, inputClasses, onChange, value }) => {
+    const _onChange = useCallback(
+        (event: ChangeEvent<HTMLTextAreaElement>) => onChange(field.name, event.target.value),
+        [field.name]
+    );
+    const className = inputClasses.textarea ?? '';
+    const _value = value === null || value === undefined ? '' : value;
+    return <textarea className={className} id={id} name={field.name} value={_value} onChange={_onChange} />;
+};
+
+const CheckboxInput: FC<AutoFormFieldProps> = ({ field, id, inputClasses, onChange, value }) => {
+    const _onChange = useCallback(
+        (event: ChangeEvent<HTMLInputElement>) => onChange(field.name, event.target.checked === true),
+        [field.name]
+    );
+    const className = inputClasses.checkbox ?? '';
+    return (
+        <input
+            className={className}
+            id={id}
+            name={field.name}
+            type="checkbox"
+            onChange={_onChange}
+            checked={value === true}
+        />
+    );
+};
+
+const SelectInput: FC<AutoFormFieldProps> = ({ field, id, inputClasses, onChange, value }) => {
+    const { name, options, placeholder } = field;
+    const _onChange = useCallback(
+        (event: ChangeEvent<HTMLSelectElement>) => {
+            const value_ = event.target.value;
+            onChange(name, value_ === '' ? null : value_);
+        },
+        [name]
+    );
+    const _value = value === null || value === undefined ? '' : value;
+    const className = inputClasses.select ?? '';
+    return (
+        <select className={className} id={id} name={name} value={_value} onChange={_onChange}>
+            {placeholder !== null && <option value="">{placeholder}</option>}
+            {options.map(option => (
+                <option key={option.value} value={option.value}>
+                    {option.label}
+                </option>
+            ))}
+        </select>
+    );
+};
+
+const RadioInput: FC<AutoFormFieldProps> = ({ field, inputClasses, onChange, value }) => {
+    const { name, options } = field;
+    const _onChange = useCallback((event: ChangeEvent<HTMLInputElement>) => onChange(name, event.target.value), [name]);
+    const className = inputClasses.radio ?? '';
+    return (
+        <div>
+            {options.map(option => (
+                <label className={className} key={option.value}>
+                    <input
+                        name={name}
```

Comment:
No id necessary because the label wraps the input, which as the same functionality as using an id and htmlFor. Plus, the ID passed in is being used by the label for the whole radio group, so if it was applied we'd get wonky behavior.